arm-software/embedded/scripts/extract_dmg.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Copyright (c) 2025, Arm Limited and affiliates.
+# Part of the Arm Toolchain project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# A bash script for extracting the contents of a MacOS .dmg image.
+
+# The script takes the following arguments:
+# extract_dmg.sh <path_to_dmg> [<glob_pattern_to_extract>] [<dst_dir>]
+# * path_to_dmg: The path to the .dmg image to extract contents from
+# * glob_pattern_to_extract (optional): A glob pattern to match the contents
+#   to be extracted. The default is "*".
+# * dst_dir (optional): The directory into which the contents should be
+#   extracted. The default is the current working dir.
+
+if [[ -z "${1}" ]]; then
+    >&2 echo "ERROR: Path to .dmg file not provided."
+    echo "usage: ${0} <path_to_dmg> [<glob_pattern_to_extract>] [<dst_dir>]"
+    exit 1
+fi
+
+DMG_PATH="${1}"
+PATTERN_TO_EXTRACT="${2:-"*"}"
+DST_DIR="${3:-"."}"
+
+if command -v hdiutil > /dev/null; then
+    ATTACH_OUTPUT=$( hdiutil attach -nobrowse -readonly "${DMG_PATH}" )
+    MOUNT_PATH=$( echo "${ATTACH_OUTPUT}" | grep '/Volumes/' | sed -r 's/.*(\/Volumes\/.*)/\1/' )
```

Comment:
Tiny sed tip: if your regex wants to match things with `/` in it, it's easier to read if you don't also use `/` as the delimiter. You can write e.g. `sed -r 's!.*(/Volumes/.*)!\1!'` and then you don't have quite so many toothpicks scattered around.